False positive too-few-public-methods for TypedDict in Python 3.9

This is a follow up to #1927, #2607, and #4072. In Python 3.9, TypedDict was changed from a class to a function with metaclass magic. This change caused worse problems initially (fixed by #4072), but now pylint gives a warning about too few public methods when subclassing TypedDict to create a new type as suggested in the documentation (https://docs.python.org/3/library/typing.html#typing.TypedDict). A work around is to use one of the alternate TypedDict syntax forms in which TypedDict is used as a function instead of a class.

### Steps to reproduce

Given a file `a.py`:
```python
"Demo"
import typing


class Coordinate(typing.TypedDict):
    "This is my type"
    x: float
    y: float
```

### Current behavior

Result of `pylint a.py`:
```
************* Module test
test.py:5:0: R0903: Too few public methods (0/2) (too-few-public-methods)

------------------------------------------------------------------
Your code has been rated at 7.50/10 (previous run: 7.50/10, +0.00)
```


### Expected behavior

No warnings.


### pylint --version output

Result of `pylint --version` output:
```
pylint 2.7.2                                                                                  
astroid 2.5.1                                                                                 
Python 3.9.2
```

--- a/ChangeLog
+++ b/ChangeLog
@@ -29,6 +29,10 @@ Release date: TBA
 
   Closes #4161
 
+* Exempt ``typing.TypedDict`` from ``too-few-public-methods`` check.
+
+  Closes #4180
+
 
 What's New in Pylint 2.7.2?
 ===========================

--- a/pylint/checkers/design_analysis.py
+++ b/pylint/checkers/design_analysis.py
@@ -95,16 +95,17 @@ SPECIAL_OBJ = re.compile("^_{2}[a-z]+_{2}$")
 DATACLASSES_DECORATORS = frozenset({"dataclass", "attrs"})
 DATACLASS_IMPORT = "dataclasses"
 TYPING_NAMEDTUPLE = "typing.NamedTuple"
+TYPING_TYPEDDICT = "typing.TypedDict"
 
 
 def _is_exempt_from_public_methods(node: astroid.ClassDef) -> bool:
     """Check if a class is exempt from too-few-public-methods"""
 
-    # If it's a typing.Namedtuple or an Enum
+    # If it's a typing.Namedtuple, typing.TypedDict or an Enum
     for ancestor in node.ancestors():
         if ancestor.name == "Enum" and ancestor.root().name == "enum":
             return True
-        if ancestor.qname() == TYPING_NAMEDTUPLE:
+        if ancestor.qname() in (TYPING_NAMEDTUPLE, TYPING_TYPEDDICT):
             return True
 
     # Or if it's a dataclass

--- a/tests/functional/t/typedDict.py
+++ b/tests/functional/t/typedDict.py
@@ -1,5 +1,5 @@
 """Test typing.TypedDict"""
-# pylint: disable=invalid-name,missing-class-docstring,too-few-public-methods
+# pylint: disable=invalid-name,missing-class-docstring
 import typing
 from typing import TypedDict
 

